src/AppInstallerCLICore/Argument.h

```diff
@@ -108,8 +140,11 @@ namespace AppInstaller::CLI
 
         // Static argument validation helpers; throw CommandException when validation fails.
 
-        // Requires that some form of package selection argument is present
-        static void ValidatePackageSelectionArgumentSupplied(const Execution::Args& args);
+        static ArgTypeCategory GetCategoriesPresent(const Execution::Args& arg);
+
+        // Requires that arguments meet common requirements
+        static ArgTypeCategory GetCategoriesAndValidateCommonArguments(const Execution::Args& args, bool requirePackageSelectionArg = true);
+        static void ValidateCommonArguments(const Execution::Args& args, bool requirePackageSelectionArg = true) { std::ignore = GetCategoriesAndValidateCommonArguments(args, requirePackageSelectionArg); }

```

Comment:
One thing I don't see as part of this PR is the concept of mutually exclusive sets (and there are multiple). I see mutually exclusive categories implemented here, which is great, but there's also things like `--rainbow`, `--retro`, and `--no-vt` which probably need to be mutually exclusive of one another. Or `--silent` and `--interactive`, `--version` and `--versions`, `--custom` and `--override`, `--purge` and `--preserve`, `--enable` and `--disable`. 

I do understand that these are probably more on a per-command basis, so it would be good to see a function that can be used in the command files where we can pass in a set of arg types, as well as the context and a potential error message to validate that only one argument within the set is present. Something like `ValidateExclusiveArguments`

Reply:
Added `ValidateExclusiveArguments()` and a single use of it in the root `Command` class for `--rainbow` and `--retro`.

I didn't look closely at all the arguments as my main motivation for this change was limiting arguments that don't make sense for multiple installs in an easy way. But I tried to make it so that it would be easy to add any rules we may be missing in the future.

Reply:
Thank you!